network: add ManageForeignRoutingPolicyRules= boolean setting

yuwata opened the pull request:
The commit 0b81225 makes that networkd remove all foreign rules except those with "proto kernel".

But, in some situation, people may want to manage routing policy rules with other tools, e.g. 'ip' command. To support such the situation, this introduce ManageForeignRoutingPolicyRules= boolean setting.

Closes #19106.

keszybz commented on man/networkd.conf.xml:
```diff
@@ -62,6 +62,15 @@
 
+      <varlistentry>
+        <term><varname>ManageForeignRoutingPolicyRules=</varname></term>
+        <listitem><para>A boolean. When true, <command>systemd-networkd</command> will store any
+        routing policy rules configured by other tools in its memory, and
+        <command>systemd-networkd</command> will remove rules that are not configured in .network files.
```
"`<command>systemd-networkd</command> will store any routing policy rules configured by other tools in its memory`" is an internal detail. Maybe just say:
"`When true, <command>systemd-networkd</command> will remove rules that are not configured in .network files (except for rules with "proto kernel"). When false, it will not remove any foreign rules, keeping them even if they are not not configured in a .network file.
`"

keszybz reviewed (commented):
LGTM. Yet-another config option is not great, but we already have ManagerForeingRoutes=, and this one is conceptually very similar. I don't see a nice way to make this automatic.

yuwata commented:
@keszybz Thank you for the review and the suggestion. The document is updated accordingly. The explanation for `ManageForeignRoutes=` is also updated based on your suggestion. One more trivial commit is also added. PTAL.
Upgrading the green label.

hligit commented:
@yuwata could this be backported to v247/v248? The removing of foreign routing rules breaks AWS-CNI https://github.com/aws/amazon-vpc-cni-k8s/issues/1600

yuwata commented:
@hligit Oh, yes, it should be. The commit 0b81225e5791f660506f7db0ab88078cf296b771 is included in v248, and backported to v247. So this PR should be also backported to them, as this is follow-ups for the commit. I will prep PRs to backport this for stable branches.

hligit commented:
@yuwata Legendary! Thanks for the swift actions!

yuwata commented:
@hligit The backporting PRs are merged.

hligit commented:
@yuwata Since the file `src/network/networkd-routing-policy-rule.c` is about routing policy rules, shouldn't `!m->manage_foreign_routes` [here](https://github.com/systemd/systemd/pull/19287/files#diff-ef75590c822c8f48cafe2408c3fdd2a7bf672b8f725b0f20e44b55bed5717093R980) be `!m->manage_foreign_rules` instead?

yuwata commented:
Ugh... Will fix soon. Thank you.

yuwata commented:
Backports:
v247: https://github.com/systemd/systemd-stable/pull/118
v248: https://github.com/systemd/systemd-stable/pull/119
v249: https://github.com/systemd/systemd-stable/pull/120
